src/legacy/core_plugins/telemetry/index.ts

```diff
@@ -89,30 +91,8 @@ const telemetry = (kibana: any) => {
         },
       },
       async replaceInjectedVars(originalInjectedVars: any, request: any) {
-        const config = request.server.config();
-        const optIn = config.get('telemetry.optIn');
-        const allowChangingOptInStatus = config.get('telemetry.allowChangingOptInStatus');
-        const currentKibanaVersion = getCurrentKibanaVersion(request.server);
-        let telemetryOptedIn: boolean | null;
-
-        if (typeof optIn === 'boolean' && !allowChangingOptInStatus) {
-          // When not allowed to change optIn status and an optIn value is set, we'll overwrite with that
-          telemetryOptedIn = optIn;
-        } else {
-          telemetryOptedIn = await getTelemetryOptIn({
-            request,
-            currentKibanaVersion,
-          });
-          if (telemetryOptedIn === null) {
-            // In the senario there's no value set in telemetryOptedIn, we'll return optIn value
-            telemetryOptedIn = optIn;
-          }
-        }
-
-        return {
-          ...originalInjectedVars,
-          telemetryOptedIn,
-        };
+        const telemetryInjectedVars = await replaceTelemetryInjectedVars(request);
+        return Object.assign(originalInjectedVars, telemetryInjectedVars);
```

Comment:
```suggestion
        return Object.assign({}, originalInjectedVars, telemetryInjectedVars);
```

nit: To avoid mutating the passed-in object.